src/dim_helpers/ConvDims.jl

```diff
@@ -81,10 +103,10 @@ function check_spdf(x_size::NTuple{N}, w_size::NTuple{N}, stride, padding, dilat
 
     # padding is kind of a special case; we allow it to be either 2-length or 4-length,
     # since we support asymmetrical padding
-    if length(ppadding) != 2*nd
+    if length(ppadding) != 2 * nd
         if length(ppadding) == nd
             # Do this repeat dance so that we get lo/hi symmetrical padding
-            ppadding = tuple(repeat(collect(ppadding), inner=2)...)
+            ppadding = NTuple{2 * nd, Int}(repeat(collect(ppadding), inner=2))
```

Comment:
I doubt it really matters but seems a shame to make small arrays when you just need tuples. How easy would it be to avoid `collect`?

Reply:
Tried implementing it with tuple-peeling, but turned out to be slower.

Reply:
Do you mean this line narrowly, or the whole thing? For just this, one attempt is:
```
julia> pad2(pp, nd=length(pp)) = NTuple{2 * nd, Int}(repeat(collect(pp), inner=2));

julia> @btime pad2($((1,2,3,)))
  min 569.745 ns, mean 600.348 ns (6 allocations, 432 bytes)
(1, 1, 2, 2, 3, 3)

julia> pad3(pp) = ntuple(i -> pp[(i-1)÷2+1], 2*length(pp));

julia> @btime pad3($((1,2,3,)))
  min 1.458 ns, mean 1.555 ns (0 allocations)
(1, 1, 2, 2, 3, 3)
```

Reply:
Much as I like 1-based indexing, I'm amazed that worked first time. But more fair timing might be:
```
julia> pad2(pp) = NTuple{2 * length(pp), Int}(repeat(collect(pp), inner=2));

julia> @btime pad2($((1,2,3,)))
  min 69.502 ns, mean 76.486 ns (2 allocations, 192 bytes)
(1, 1, 2, 2, 3, 3)
```

Reply:
Didn't think about using `ntuple` function... :) Using it right now.